File: apk/app/src/main/java/com/example/momentra/ui/shell/shared/TravelCurrencyPicker.kt
package com.example.momentra.ui.shell.shared

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.momentra.R
import com.example.momentra.ui.shell.group.shared.TravelCurrencyCatalog
import com.example.momentra.ui.theme.PlusJakartaSans

@Composable
fun TravelCurrencyPickerRow(
    selectedCode: String,
    onSelected: (String) -> Unit,
    preferredCodes: List<String>,
    modifier: Modifier = Modifier,
    label: String = "Currency",
    textColor: Color = Color(0xFFE5E0EE),
    secondaryColor: Color = Color(0xFFC9C4D8),
    accentColor: Color = Color(0xFF14B8A6),
    labelFontSize: TextUnit = 12.sp,
    valueFontSize: TextUnit = 14.sp,
    symbolFontSize: TextUnit = 28.sp,
    cornerRadius: Dp = 12.dp,
    showLabel: Boolean = true,
) {
    val options = remember(preferredCodes) { MomentCurrencyResolver.pickerOptions(preferredCodes) }
    var menuOpen by remember { mutableStateOf(false) }
    val symbol = TravelCurrencyCatalog.symbol(selectedCode)

    Box(modifier = modifier) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .clickable { menuOpen = true }
                .padding(horizontal = 4.dp, vertical = 2.dp),
            horizontalArrangement = if (showLabel) Arrangement.SpaceBetween else Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            if (showLabel) {
                Text(label, color = secondaryColor, fontSize = labelFontSize, fontFamily = PlusJakartaSans)
            }
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(6.dp),
            ) {
                Text(
                    symbol,
                    color = accentColor,
                    fontSize = symbolFontSize,
                    fontWeight = FontWeight.Bold,
                    fontFamily = PlusJakartaSans,
                )
                Text(
                    selectedCode,
                    color = textColor,
                    fontSize = valueFontSize,
                    fontWeight = FontWeight.SemiBold,
                    fontFamily = PlusJakartaSans,
                )
                Icon(
                    painter = painterResource(R.drawable.ic_biz_create_chevron),
                    contentDescription = null,
                    tint = secondaryColor,
                    modifier = Modifier.size(16.dp),
                )
            }
        }
        DropdownMenu(expanded = menuOpen, onDismissRequest = { menuOpen = false }) {
            options.forEach { code ->
                DropdownMenuItem(
                    text = {
                        Text(
                            TravelCurrencyCatalog.display(code),
                            fontFamily = PlusJakartaSans,
                        )
                    },
                    onClick = {
                        onSelected(code)
                        menuOpen = false
                    },
                )
            }
        }
    }
}

File: momentra/momentra/Shell/Shared/TravelCurrencyPicker.swift
import SwiftUI

struct TravelCurrencyPicker: View {
    @Binding var selectedCode: String
    let preferredCodes: [String]
    var label: String = "Currency"
    var showLabel: Bool = true
    var textColor: Color = Color(hex: "#E5E0EE")
    var secondaryColor: Color = Color(hex: "#C9C4D8")
    var accentColor: Color = Color(hex: "#14B8A6")

    private var options: [String] {
        MomentCurrencyResolver.pickerOptions(preferred: preferredCodes)
    }

    var body: some View {
        Menu {
            ForEach(options, id: \.self) { code in
                Button(TravelCurrencyCatalog.display(code)) { selectedCode = code }
            }
        } label: {
            HStack {
                if showLabel {
                    Text(label)
                        .font(.system(size: 12, weight: .semibold))
                        .foregroundStyle(secondaryColor)
                    Spacer()
                }
                HStack(spacing: 6) {
                    Text(TravelCurrencyCatalog.symbol(selectedCode))
                        .font(.system(size: showLabel ? 14 : 28, weight: .bold))
                        .foregroundStyle(accentColor)
                    Text(selectedCode)
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundStyle(textColor)
                    Image(systemName: "chevron.down")
                        .font(.system(size: 12))
                        .foregroundStyle(secondaryColor)
                }
            }
            .padding(12)
            .background(Color(hex: "#201E28"))
            .overlay(RoundedRectangle(cornerRadius: 12).stroke(Color(hex: "#938EA1"), lineWidth: 1))
            .clipShape(RoundedRectangle(cornerRadius: 12))
        }
    }
}
